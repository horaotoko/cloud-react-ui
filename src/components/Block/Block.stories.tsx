import { Story, Meta } from '@storybook/react';
import { Block, Props } from './Block';

export default {
  component: Block,
  title: 'components/Block',
} as Meta;

const Template: Story<Props> = (args) => <Block {...args}>Block</Block>;

export const Default = Template.bind({});

export const Bordered = Template.bind({});
Bordered.args = { border: true };
