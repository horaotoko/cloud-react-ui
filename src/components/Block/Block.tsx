import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";

export type Props = {
  border?: boolean;
  children: React.ReactNode;
  className?: string;
};

const StyledBlock = styled.div<Props>`
  ${({ theme: { block }, border = false }) => css`
    width: ${block.width};
    margin: ${block.margin};
    padding: ${block.padding};
    background: ${block.background};
    border-radius: ${block.borderRadius};
    border: ${border ? block.border : undefined};
    text-align: ${block.textAlign};
    box-sizing: border-box;
  `}
`;
StyledBlock.defaultProps = defaultProps;

const Block = forwardRef<HTMLDivElement, Props>((props, ref) => (
  <StyledBlock ref={ref} {...props} />
));

Block.displayName = "Block";

export { Block };
