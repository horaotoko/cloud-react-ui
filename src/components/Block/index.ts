export { Block } from "./Block";
export type { Props as BlockProps } from "./Block";
