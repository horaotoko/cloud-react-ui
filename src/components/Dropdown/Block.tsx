import styled from "styled-components";

export const Block = styled.div`
  padding: 16px;

  & + & {
    border-top: 1px solid #d4d8dd;
  }
`;

Block.displayName = "Dropdown.Block";
