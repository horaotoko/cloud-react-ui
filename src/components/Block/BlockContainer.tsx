import { BlockPresenter } from "./BlockPresenter";

export type BlockProps = {
  border?: boolean;
  className?: string;
};

export const BlockContainer: React.FC<BlockProps> = (props) => {
  return <BlockPresenter {...props} />;
};
