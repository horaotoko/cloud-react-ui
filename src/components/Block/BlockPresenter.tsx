import styled, { css } from "styled-components";
import { BlockProps } from "./BlockContainer";
import { defaultProps } from "../../theme";

const StyledBlock = styled.div`
  ${({ theme: { block } }) => css`
    width: ${block.width};
    margin: ${block.margin};
    padding: ${block.padding};
    background: ${block.background};
    border-radius: ${block.borderRadius};
    border: ${block.border};
    text-align: ${block.textAlign};
    box-sizing: border-box;
  `}
`;
StyledBlock.defaultProps = defaultProps;

export const BlockPresenter: React.FC<BlockProps> = (props) => {
  return <StyledBlock {...props} />;
};
