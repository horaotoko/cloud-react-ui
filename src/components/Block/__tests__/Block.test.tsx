import { render } from '@testing-library/react';
import { Block } from '../Block';

describe('Block', () => {
  it('snapshot test', () => {
    const { asFragment } = render(<Block>block</Block>);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('border props', () => {
    it('border=true', () => {
      const { container } = render(<Block border>block</Block>);
      expect(container.firstChild).toHaveStyle('border: 1px solid #d4d8dd;');
    });

    it('border=false', () => {
      const { container } = render(<Block>block</Block>);
      expect(container.firstChild).toHaveStyle('border: none;');
    });
  });
});
