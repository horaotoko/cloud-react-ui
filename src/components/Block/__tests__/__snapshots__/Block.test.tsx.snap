// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block snapshot test 1`] = `
<DocumentFragment>
  .c0 {
  background: #FFFFFF;
  border-radius: 4px;
  box-sizing: border-box;
}

<div
    class="c0"
  >
    block
  </div>
</DocumentFragment>
`;
